detectron2/structures/boxes.py

```diff
@@ -311,33 +311,49 @@ def __iter__(self):
         yield from self.tensor
 
 
-# implementation from https://github.com/kuangliu/torchcv/blob/master/torchcv/utils/box.py
-# with slight modifications
-def pairwise_iou(boxes1: Boxes, boxes2: Boxes) -> torch.Tensor:
+def intersection_cacl(
+    boxes1: Boxes, boxes2: Boxes
+) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
     """
     Given two lists of boxes of size N and M,
-    compute the IoU (intersection over union)
-    between __all__ N x M pairs of boxes.
-    The box order must be (xmin, ymin, xmax, ymax).
-
+    compute their independent boxes areas and intersection areas.
     Args:
         boxes1,boxes2 (Boxes): two `Boxes`. Contains N & M boxes, respectively.
 
     Returns:
-        Tensor: IoU, sized [N,M].
+        Tensor: intersection, sized [N,M].
+        Tensor: area1, area2, sized N and M
```

Comment:
Computing both of them is unnecessary for `pairwise_ioa`.
areas can be computed in `pairwise_iou/ioa` when needed and this function will compute intersection only. This also simplifies the interface of this function.